portfolio/src/main/webapp/script.js

```diff
@@ -60,33 +60,55 @@ async function displayComments() {
 }
 
 /**
- * Displays the login status of the user and a link for logging in or out.
+ * Retrieves the login information for the user.
  */
-async function displayLoginInfo() {
+async function getLoginInfo() {
   const response = await fetch('/login');
   const loginInfo = await response.json();
-  const loginUrl = loginInfo[0];
-  const logoutUrl = loginInfo[1];
-  const userEmail = loginInfo[2];
 
+  return loginInfo;
+}
+
+/**
+ * Displays the login status of the user and a link for logging in or out.
+ */
+function displayLoginInfo(loginInfo) {
   const loginContainer = document.getElementById('login-container');
 
   // Check if the user is logged in.
-  if (userEmail === null) {
+  if (!loginInfo.isLoggedIn) {
```

Comment:
I would try to avoid 
```
if (not condition) { 
  A;  
} else { 
  B; 
}
```

Because the 'else' clause will read as 'not not condition'

So instead I would do 
```
if (condition) { 
  B;  
} else { 
  A; 
}
```